File: app/src/main/java/com/farionik/yandextestapp/data/BaseDao.java
package com.farionik.yandextestapp.data;

import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Update;

import java.util.List;

public interface BaseDao<T> {
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insert(T entity);

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insertAll(List<T> entities);

    @Insert(onConflict = OnConflictStrategy.IGNORE)
    void insertIgnore(T entity);

    @Insert(onConflict = OnConflictStrategy.IGNORE)
    void insertAllIgnore(List<T> entities);

    @Update
    void update(T entity);

    @Update(onConflict = OnConflictStrategy.IGNORE)
    void updateIgnore(T entity);
}

File: app/src/main/java/com/farionik/yandextestapp/data/BaseDao.kt
package com.farionik.yandextestapp.data

import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Update

interface BaseDao<T> {
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    suspend fun insert(entity: T)

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    suspend fun insertAll(entities: List<T>)

    @Insert(onConflict = OnConflictStrategy.IGNORE)
    suspend fun insertIgnore(entity: T)

    @Insert(onConflict = OnConflictStrategy.IGNORE)
    suspend fun insertAllIgnore(entities: List<T>)

    @Update
    suspend fun update(entity: T)

    @Update(onConflict = OnConflictStrategy.IGNORE)
    suspend fun updateIgnore(entity: T)
}